src/ripple/app/tx/impl/CFTokenIssuanceCreate.cpp

```diff
@@ -0,0 +1,127 @@
+//------------------------------------------------------------------------------
+/*
+  This file is part of rippled: https://github.com/ripple/rippled
+  Copyright (c) 2023 Ripple Labs Inc.
+
+  Permission to use, copy, modify, and/or distribute this software for any
+  purpose  with  or without fee is hereby granted, provided that the above
+  copyright notice and this permission notice appear in all copies.
+
+  THE  SOFTWARE IS PROVIDED "AS IS" AND THE AUTHOR DISCLAIMS ALL WARRANTIES
+  WITH  REGARD  TO  THIS  SOFTWARE  INCLUDING  ALL  IMPLIED  WARRANTIES  OF
+  MERCHANTABILITY  AND  FITNESS. IN NO EVENT SHALL THE AUTHOR BE LIABLE FOR
+  ANY  SPECIAL ,  DIRECT, INDIRECT, OR CONSEQUENTIAL DAMAGES OR ANY DAMAGES
+  WHATSOEVER  RESULTING  FROM  LOSS  OF USE, DATA OR PROFITS, WHETHER IN AN
+  ACTION  OF  CONTRACT, NEGLIGENCE OR OTHER TORTIOUS ACTION, ARISING OUT OF
+  OR IN CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
+*/
+//==============================================================================
+
+#include <ripple/app/tx/impl/CFTokenIssuanceCreate.h>
+#include <ripple/ledger/View.h>
+#include <ripple/protocol/Feature.h>
+#include <ripple/protocol/TxFlags.h>
+#include <ripple/protocol/st.h>
+
+namespace ripple {
+
+NotTEC
+CFTokenIssuanceCreate::preflight(PreflightContext const& ctx)
+{
+    if (!ctx.rules.enabled(featureCFTokensV1))
+        return temDISABLED;
+
+    if (auto const ret = preflight1(ctx); !isTesSuccess(ret))
+        return ret;
+
+    if (ctx.tx.getFlags() & tfCFTokenIssuanceCreateMask)
+        return temINVALID_FLAG;
+
+    if (auto const fee = ctx.tx[~sfTransferFee])
+    {
+        if (fee > maxTransferFee)
+            return temBAD_CFTOKEN_TRANSFER_FEE;
+
+        // If a non-zero TransferFee is set then the tfTransferable flag
+        // must also be set.
+        if (fee > 0u && !ctx.tx.isFlag(tfTransferable))
+            return temMALFORMED;
+    }
+
+    // TODO length?
+    // if (auto const code = ctx.tx[~sfAssetCode]; code.length() >
+    // maxAssetCodeLength)
+    //     return temMALFORMED;
+
+    if (auto const metadata = ctx.tx[~sfCFTokenMetadata])
+    {
+        if (metadata->length() == 0 ||
+            metadata->length() > maxCFTokenMetadataLength)
+            return temMALFORMED;
+    }
+
+    return preflight2(ctx);
+}
+
+TER
+CFTokenIssuanceCreate::preclaim(PreclaimContext const& ctx)
+{
+    // TODO - check number of CFTs? not sure why we need to limit it
+
+    // if already a CFT with this asset code - error
+    // TODO - need to check this again before insert too?
+    if (ctx.view.read(
+            keylet::cft_issuance(ctx.tx[sfAccount], ctx.tx[sfAssetCode])))
```

Comment:
Small thing, but there's an `exists` method on `ReadView` that might be _marginally_ more performant here

Reply:
Just seeing that this comment is resolved but the PR doesn't seem to have changed -- is there an update forthcoming @ledhed2222 ?